tools/node_tcp.js

```diff
@@ -0,0 +1,17 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+const { Server } = require("net");
+const port = process.argv[2] || "4544";
+console.log("port", port);
+
+const response = Buffer.from(
+  "HTTP/1.1 200 OK\r\nContent-Length: 12\r\n\r\nHello World\n"
+);
+
+Server(socket => {
+  socket.on("data", _ => {
+    socket.write(response);
```

Comment:
Should be `.end()` or this leaks sockets?
Or are you just profiling one tcp stream?

Reply:
we're profiling keep-alive - so it shouldn't end here.

Reply:
The command we use to benchmark - `third_party/wrk/linux/wrk -d 10s http://127.0.0.1:4544/`, creates 10 connections, thus 10 sockets are created to reply simultaneously (thought in practise this is creating 11). We can confirm this by adding those lines:

```diff
+ let i = 0;
  Server(socket => {
+   console.log(`Connected to socket no ${++i}`)
    socket.on("data", _ => {
    socket.write(response);
```

Which will log:

```
Connected to socket no 1
Connected to socket no 2
Connected to socket no 3
Connected to socket no 4
Connected to socket no 5
Connected to socket no 6
Connected to socket no 7
Connected to socket no 8
Connected to socket no 9
Connected to socket no 10
Connected to socket no 11
```

Reply:
> we're profiling keep-alive - so it shouldn't end here.

Perhaps a comment to this effect will prevent it being "fixed" later.

Reply:
I second @hayd 's suggestion.

Reply:
Fixed.